.github/workflows/ci-cd.yml

```diff
@@ -0,0 +1,267 @@
+name: Build->Validate->Deploy
+
+concurrency: 
+  group: ${{ github.ref }}
+  cancel-in-progress: true
+
+on:
+  workflow_dispatch:
+  pull_request:
+    paths-ignore:
+      - '**.md'
+  push:
+    branches:
+      - master
+    paths-ignore:
+      - '**.md'
+  schedule:
+    # runs the CI everyday at 10AM
+    - cron: "0 10 * * *"
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    strategy:
+      # fail-fast: true
+      matrix:
+        channel:
+          - stable
+
+    steps:
+      - uses: actions/checkout@v3.1.0
+        
+      - uses: subosito/flutter-action@v2.7.1
+        with:
+          channel: ${{ matrix.channel }}
+
+      - name: Add pub cache bin to PATH
+        run: echo "$HOME/.pub-cache/bin" >> $GITHUB_PATH
+
+      - name: Add pub cache to PATH
+        run: echo "PUB_CACHE="$HOME/.pub-cache"" >> $GITHUB_ENV
+
+      - run: dart pub global activate melos
```

Comment:
There is a melos action that can be used:
https://github.com/marketplace/actions/melos-action

Reply:
Thanks for letting me know! I'll look into it once I have some free time to work on this PR.